testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCheckpointsCompatibility.java

```diff
@@ -378,6 +380,226 @@ private void testCheckpointNullStatisticsForRowType(Consumer<String> sqlExecutor
+    {
+        onDelta().executeQuery(format(
+                "CREATE TABLE default.%s" +
+                        "(a_number INT, a_string STRING) " +
+                        "USING DELTA " +
+                        "PARTITIONED BY (a_number) " +
+                        "LOCATION 's3://%s/databricks-compatibility-test-%1$s' " +
+                        "TBLPROPERTIES (" +
+                        " delta.checkpointInterval = 5, " +
+                        " delta.checkpoint.writeStatsAsJson = false)",
+                tableName, bucketName));
+
+        try {
+            sqlExecutor.accept("INSERT INTO " + qualifiedTableName + " VALUES (1,'ala')");
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
+                    .containsOnly(ImmutableList.of(
+                            row("a_number", null, 1.0, 0.0, null, null, null),
+                            row("a_string", null, null, 0.0, null, null, null),
+                            row(null, null, null, null, 1.0, null, null)));
+        }
+        finally {
+            onDelta().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoWriteStatsAsStructDisabled()
+    {
+        String tableName = "test_dl_checkpoints_write_stats_as_struct_disabled_trino_" + randomTableSuffix();
+        testWriteStatsAsStructDisabled(sql -> onTrino().executeQuery(sql), tableName, "delta.default." + tableName);
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    public void testDatabricksWriteStatsAsStructDisabled()
+    {
+        String tableName = "test_dl_checkpoints_write_stats_as_struct_disabled_databricks_" + randomTableSuffix();
+        testWriteStatsAsStructDisabled(sql -> onDelta().executeQuery(sql), tableName, "default." + tableName);
+    }
+
+    private void testWriteStatsAsStructDisabled(Consumer<String> sqlExecutor, String tableName, String qualifiedTableName)
+    {
+        onDelta().executeQuery(format(
+                "CREATE TABLE default.%s" +
+                        "(a_number INT, a_string STRING) " +
+                        "USING DELTA " +
+                        "PARTITIONED BY (a_number) " +
+                        "LOCATION 's3://%s/databricks-compatibility-test-%1$s' " +
+                        "TBLPROPERTIES (" +
+                        " delta.checkpointInterval = 1, " +
+                        " delta.checkpoint.writeStatsAsJson = false, " + // Disable json stats to avoid merging statistics with 'stats' field
+                        " delta.checkpoint.writeStatsAsStruct = false)",
+                tableName, bucketName));
+
+        try {
+            sqlExecutor.accept("INSERT INTO " + qualifiedTableName + " VALUES (1,'ala')");
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
```

Comment:
We are checking the compatibility with Databricks here.

Let's check that the stats we are writing can be indeed read by Databricks as well.
```
"DESC EXTENDED default." + tableName + " " + columnName
```

Reference: https://docs.microsoft.com/en-us/azure/databricks/spark/latest/spark-sql/language-manual/sql-ref-syntax-aux-analyze-table

Reply:
Databricks doesn't return column statistics unless executing `ANALYZE TABLE`. I can add such assertions, but it seems not so helpful.

Reply:
> if the stats exist in the checkpoint (created by Trino) , but we don't explicitly do ANALYZE TABLE on Databricks, the stats are not available?

The stats is shown in Trino `SHOW STATS` and not shown in Spark `DESC EXTENDED`. The behavior is same as Spark writes. 

> What if we do the ANALYZE TABLE with onDelta() when the table is empty, and then we create the checkpoint with Trino? Are the stats still not available?

The stats are still unavailable, old stats will be returned. This behavior is also same as Spark writes.